### DISCUSSION TITLE
How to set parameter number_of_carriers and number_of_shells in simulate_waveforms()

### DISCUSSION THREAD

**Discussion by zql2021** [JuliaPhysics/SolidStateDetectors.jl#378] [Q&A] [answered]
Hello,
When using simulate_waveforms(), I want to take into account the **charge cloud** effect, so I set both` diffusion `and `self_repulsion` to` true`. However, I'm unclear about how to set `number_of_carriers and number_of_shells.` Even though I know these two parameters are eventually passed into the NBodyChargeCloud().
And `number_of_carriers `means Number of charge carriers to be used in the N-Body simulation of an energy deposition.
`number_of_shells` means Number of shells around the center point of the energy deposition.

Could you give me some advice on how to set number_of_carriers and number_of_shells?
Thank you very much and looking forward to your reply.

**Comment by fhagemann** [comment, marked as answer]
These things you have to decide on yourself.
The bigger `number_of_carriers`, the smoother the simulated output might be. But the simulation will also take significantly longer.

One thing to note: the energy deposited **per change carrier** in the outer shells in smaller than in the inner shells, because there are more charge carriers on the outer shells than inside.

**Reply by zql2021** [reply]
I get it now. Thank you for your quick response.
